server.js

```diff
@@ -1,35 +1,116 @@
+  .then(() => {
+    console.log("Successfully connected to MongoDB");
+  })
+  .catch((error) => {
+    console.error("Error connecting to MongoDB:", error.message);
+    process.exit(1); // Exit if connection fails
+  });
 
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
-mongoose.connect(mongoUrl);
 mongoose.Promise = Promise;
 
-// Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
-// PORT=9000 npm start
+// Define the port the app will run on, defaults 8080
 const port = process.env.PORT || 8080;
+console.log(process.env.PORT);
 const app = express();
 
+// Define book model using Mongoose schema
+const Book = mongoose.model("Book", new mongoose.Schema({
+  bookID: Number,
+  title: String,
+  authors: String,
+}));
+
+// Seed the database
+if (process.env.RESET_DB) {
+  const seedDatabase = async () => {
+    await Book.deleteMany({}); // Clear existing database
+
+    booksData.forEach((bookData) => {
+      new Book(bookData).save(); //Save each book 
+    });
+
+    console.log("Database seeded with book data!");
+  };
+  seedDatabase(); // Call to seed database
+}
+
 // Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
 
-// Start defining your routes here
+// Define routes for API
+// Home route to list all available endpoints
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  const endpoints = listEndpoints(app); // Automaticlly list all endpoints
+  res.json({
+    message: "Hello and welcome to the libray! Here are all the endpoints:",
+    endpoints: endpoints
+  });
+});
+
+// Route to fetch all books
+app.get("/books", async (req, res) => {
+  const books = await Book.find();
+  res.json(books);
+});
+
+// Route to fetch details for a single book by its ID
+app.get("/books/:bookID", async (req, res) => {
+  const { bookID } = req.params;
+  
+  try {
+    const book = await Book.findOne({ bookID: bookID }); // Find a book by bookID
+
+    if (!book) {
+      return res.status(404).json({ message: "Book not found" }); // Handle not found
+    }
+
+    res.json(book); // Send book details
+  } catch (error) {
+    res.status(500).json({ message: error.message }); // Handle server errors
+  }
+});
+
+// Route to fetch book by author
+app.get("/books/authors/:author", async (req, res) => {
+  const { author } = req.params;
+  try {
+    // Use a case-insensitive regex to match authors
+    const books = await Book.find({ authors: new RegExp(author, "i") });
+
+    if (books.length === 0) {
+      return res.status(404).send("No books found for this author"); // Handle no results
+    }
+    res.json(books); // Send list of books by the author
+  } catch(error) {
+    console.error("Error retrieving books by author", error); // Log error
+    res.status(500).send("Server error"); // Handle server errors
+  }
 });
```

Comment:
Remember that even though you're filtering on an author, you're still returning _books_. So it would be more RESTful to make this a query param under the /books endpoint. Remember for future!